src/components/Overlay/index.tsx

```diff
@@ -58,12 +59,20 @@ class OverlayProvider extends React.Component<
   }
 }
 
-export const Overlay: React.SFC<{ context: OverlayContextInterface }> = ({
+interface OverlayProps {
+  context: OverlayContextInterface;
+  className?: string;
+}
+export const Overlay: React.SFC<OverlayProps> = ({
   children,
+  className,
   context: { type, theme, hide }
 }) => (
   <div
-    className={`overlay${type ? ` overlay--${type}` : ""}`}
+    className={classNames(
+      `overlay${type ? ` overlay--${type}` : ""}`,
+      className
+    )}
```

Comment:
Maybe:

```typescript
    className={classNames({
        overlay,
        [`overlay--${type}`]: type
        className
    })}
```

Reply:
Uh, I was lazy and copied code written before 😞